### PR TITLE
chore(concurrency): add return values to WithLock/WithRLock

### PR DESCRIPTION
**Pull request by dhaus67**
## Description

Based [on a discussion within the style guide](https://github.com/stackrox/stackrox/pull/7282#discussion_r1290644635) regarding improving the usage of `WithLock/RLock` to allow multiple return values.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

- see CI.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by dhaus67** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7993** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/7993" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/7993?utm_source=stack-comment).

File: central/sensor/service/connection/sensorevents.go [hunk 1]
```diff
@@ -114,9 +114,8 @@ func (s *sensorEventHandler) addMultiplexed(ctx context.Context, msg *central.Ms
 	metrics.IncSensorEventsDeduper(false, msg)
 
 	// Lazily create the queue for a type when not found.
-	var queue *workerQueue
-	concurrency.WithRLock(&s.workerQueuesMutex, func() {
-		queue = s.workerQueues[workerType]
+	queue := concurrency.WithRLock1(&s.workerQueuesMutex, func() *workerQueue {
+		return s.workerQueues[workerType]
```
Comment (0x656b694d) [inline central/sensor/service/connection/sensorevents.go:118]: Why this, if lines 122-123?

**Review by connorgorman — APPROVED** [review]
Please just fix the ordering to avoid an unknown side effect, but looks good

**Pull request merged**